### DISCUSSION TITLE
Dark theme

### DISCUSSION THREAD

**Discussion by vit1251** [jmjuanes/siimple#176] [Q&A]
How to switch siimple on dark theme?

**Comment by jmjuanes** [comment]
Hello! Unfortunately, at this moment we do not support dark mode toggle, but is something that we can plan for future releases. 

But in the meantime, you can customize siimple to use a darker theme, for example: 

```js
import colors from "@siimple/colors";
import base from "@siimple/preset-base";

export default {
    ...base,
    colors: {
        ...base.colors,
        background: colors.gray["800"],
        text: "white",
    },
    // ...other config
};
```
 
You can try this in [our playground app](https://siimple.xyz/playground#html=DwEwlgbgBAxgNgQwM5ILwCIxILQDM4CmAHlABbLbgBOBMALmAPYB22MjcArgLbNkVg6BbjhgFmQqvxwArTkga4Anm3GTp2AO55OcOBtLYAnADZ0APgBQUG1FCRYiFBnI5tAZgBMB7F40AjAHNsODBA0jooLGwABzA9C2tbZPtoeGQ0dA1tXF19V2xDXL0onHxiDUFhUTUCKQK5BTBlVQk6xOTO22AwRwyMJDBsQbBuGMINQYAvAmwADg0hIjpKBCoAawtgAHowKy6bHfAIfe7t49PD477ndALuFZNJsBnfbOMngvH5jRiqH4KSxWYjaVA6BwAKqQCHZ%2FOZuGBmKMEHAdnCoAhmCBYeY6NDuAQEP5CGjzFAAMIAZUpUDojA460EADokodzpBLnZrulbvcVt5okhhIIqJikDE1mpwV1NLKmSMxoQmUQlFNWXZ2SdLEcOZYgA%3D%3D&config=PTAEE0HsFdQYwIYDtQEsC2AHSAnALqDDvJADa6iYKkCmeeNoek8ONCDoAnkUwBY10NAFAg0WXAThlcAZ1AAzHJHSgARAAFZqDJlrBp5HLLUBuYaLBRYiFLsmU2suvOagaADwZIAJqG26tKDQ2kgA5qBsIQgARkGyeFy08sh%2BeAJCssL2%2BCRG8koq6lo6WPqGcmbZErkxCM6KyqqaAWU0wJhOdAC0dc5Vwp7YuT40CgjQpAQA3sKg86AAdMt9NAA0cwsVxgBcoLMLh0sr9TSL27IbRwt1cADWYcrQvnsXi48IXADaagAcAAz%2FNQAXSu1yYnjwezUAHc%2BKgGGowfMAL7I0BiZaLSDpGjEaRIBSoMLCFHmIA%3D%3D%3D).

**Comment by marshmanGit** [comment]
Just sharing in hopes this info is helpful.   As you know, Light / Dark theme selection on the OS is more popular than ever.  Instead of having your site perform a theme toggle ( which it sounds like you may be attempting ), you should start using the @media query so the users OS setting will be used.

For example:
body {
      background-color: var(--color-bg);
      color: var(--color-fg);
}

@media (prefers-color-scheme: light) {
    :root {
        --color-bg: #ffffff;
        --color-fg: #151515;
   }
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-bg: #151515;
    --color-fg: #ffffff;
  }
}

I recently did this for a material theme and only ended up with a small handful of actual color swapping that was required to make the dark theme more appropriate with the css default styles.  I have used siimple as well and recall a good number of the default styles look good by default so I suspect only a handful of tweaks will be needed.

**Comment by jmjuanes** [comment]
I have created the two issues needed to enable dark theme toggle in siimple:

- https://github.com/jmjuanes/siimple/issues/177 
- https://github.com/jmjuanes/siimple/issues/178

For the color modes, there are two options:
- Use the `media (prefers-color-scheme: dark)` rule to switch to a dark/light theme using the system settings, as described by @marshmanGit (thank you :raised_hands:).
- Use a `data-color-mode` attribute in the `<html>` tag to decide which color mode should be applied.
